ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/monitor-sessions.j2

```diff
@@ -0,0 +1,67 @@
+{# Leaf connected_endpoint monitor-sessions #}
+{% if monitor_sessions_data | length > 0 %}
+monitor_sessions:
+{%     for sname, group in monitor_sessions_data | groupby("name") %}
+{%         set group_roles = group | map(attribute='role', default='notdefined') | list %}
+{%         if 'source' in group_roles and 'destination' in group_roles %}
+  - name: {{ sname }}
+    sources:
+{%             for session in group | selectattr('role', 'equalto', 'source') %}
+{%                 if session.interfaces is arista.avd.defined %}
+      - name: {{ session.interfaces | join(", ") }}
+{%                 endif %}
+{%                 if session.source_settings is arista.avd.defined %}
+{%                     if session.source_settings.direction is arista.avd.defined %}
+        direction: {{ session.source_settings.direction }}
+{%                     endif %}
+{%                     if session.source_settings.access_group is arista.avd.defined %}
+        access_group:
+{%                         if session.source_settings.access_group.type is arista.avd.defined %}
+          type: {{ session.source_settings.access_group.type }}
+{%                         endif %}
+{%                         if session.source_settings.access_group.name is arista.avd.defined %}
+          name: {{ session.source_settings.access_group.name }}
+{%                         endif %}
+{%                         if session.source_settings.access_group.priority is arista.avd.defined %}
+          priority: {{ session.source_settings.access_group.priority }}
+{%                         endif %}
+{%                     endif %}
+{%                 endif %}
+{%             endfor %}
+    destinations:
+{%             set merged_destinations = group | selectattr('role', 'equalto', 'destination') | combine(list_merge='append') %}
```

Comment:
Have you tested this? I wouldn't think this would work in the combine filter.
Maybe call it `merged_settings` since you use it for all other settings than source?

Reply:
I did a test, its in the molecule. I'm merging:

```
server19_monitoring_session_destination:
    rack: RackC
    adapters:
      - endpoint_ports: [ Eth1 ]
        switch_ports: [ Ethernet27, Ethernet28 ]
        switches: [ DC1-SVC3A ]
        monitor_sessions:
          - name: MonitoringSessionServer18WithDest
            role: destination
            session_settings:
              encapsulation_gre_metadata_tx: true
              header_remove_size: 20
              access_group:
                type: ip
                name: ip_acl
              rate_limit_per_ingress_chip: 30 bps
              rate_limit_per_egress_chip: 30 bps
              sample: 10
              truncate:
                enabled: true
                size: 20

  server20_monitoring_session_destination:
    rack: RackC
    adapters:
      - endpoint_ports: [ Eth1 ]
        switch_ports: [ Ethernet40, Ethernet41 ]
        switches: [ DC1-SVC3A ]
        monitor_sessions:
          - name: MonitoringSessionServer18WithDest
            role: destination
            session_settings:
              encapsulation_gre_metadata_tx: false
              header_remove_size: 200
              access_group:
                type: mac
                name: mac_acl
              rate_limit_per_ingress_chip: 30 bps
              rate_limit_per_egress_chip: 30 bps
              sample: 10
              truncate:
                enabled: true
                size: 20
```

All the switchports here are destinations for the same monitoring_session (identified by name). Idea was to "append" lists (to get all interfaces from both definitions), but merge the dict (the global_settings) with default replace and not recursive (to avoid mixing between settings in the nested stuff). This is not ideal, but I don't see a better way to do it, because defining different global_settings is more like an "invalid user input" anyway?

Reply:
I would do recursive merging. And I believe you need this syntax for it to work: 
```jinja
{% {} | combine([group], recursive=true, list_merge=append) %}
```